### PR TITLE
Added query to DB

### PR DESCRIPTION
**Pull request by pashacbk**
implemented dagger to GroupInfoActivity
added methods to transform response from DB and set to View
removed commented lines
extract strings to constant variables

HomeActivity RX is on Alexander's branch

### CODE REVIEW

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivity.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    @BindView(R.id.group_status)
+    TextView status;
+
+    @Inject
+    GroupInfoActivityPresenter presenter;
+
+    public static void newIntent(Context context, int groupId) {
+        Intent intent = new Intent(context, GroupInfoActivity.class);
+        intent.putExtra(GROUP_ID, groupId);
+        context.startActivity(intent);
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivity.java]: return the created intent, don't start the activity here

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivity.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_group_info);
+        ButterKnife.bind(this);
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivity.java]: move to baseActivity

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Override
+    protected void onViewAttached() {
+        super.onViewAttached();
+        addDisposable(onSetUpView());
+    }
+
+    Disposable onSetUpView() {
+        return model.getGroupFromDb(getView().getGroupId())
+                .subscribe(group1 -> setUpView(group1));
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java]: rename group1

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/CardListAdapter.java [hunk 4]
```diff
@@ -27,13 +35,15 @@ public void swapData(List<Group> cardList) {
...
-        SingleCardHolder cardHolder = new SingleCardHolder(v);
-        return cardHolder;
+        return new SingleCardHolder(v);
+
     }
 
     @Override
     public void onBindViewHolder(SingleCardHolder singleCardHolder, int position) {
         singleCardHolder.bind(cardList.get(position));
+        singleCardHolder.itemView.setOnClickListener(i ->
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/CardListAdapter.java]: rename i

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java [hunk 5]
```diff
@@ -16,13 +16,17 @@
...
+
 @HomeActivityScope
 public class GroupListAdapter extends RecyclerView.Adapter<GroupListAdapter.GroupHolder> {
 
     private List<SingleGroupModel> groupList = new ArrayList<>();
+    private PublishSubject<Integer> groupPublishSubject = PublishSubject.create();
 
     @Inject
-    public GroupListAdapter() {}
+    GroupListAdapter() { }
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java]: are you injecting in an empty method? what is this?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java [hunk 6]
```diff
@@ -33,14 +37,19 @@ public void setGroupList(List<SingleGroupModel> groupList) {
...
 
     @Override
     public void onBindViewHolder(GroupHolder groupHolder, int position) {
         groupHolder.bind(groupList.get(position));
     }
 
+    public Observable<Integer> onGroupClick() {
+        return groupPublishSubject;
+    }
+
```
Comment (CebotariM) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java]: extra spaces

File: MeetUp/app/src/main/res/layout/activity_group_info.xml [hunk 7]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:fitsSystemWindows="true"
+    tools:context=".screens.group_screen.GroupInfoActivity">
```
Comment (CebotariM) [inline MeetUp/app/src/main/res/layout/activity_group_info.xml]: what does tools:context do?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java (hunk 5 above)
Comment (pashacbk) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/GroupListAdapter.java]: Inject GroupListAdapter in Homeactivity

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/CardListAdapter.java [hunk 8]
```diff
@@ -54,7 +64,6 @@ public int getItemCount() {
             this.cardImageId = view.findViewById(R.id.group_image);
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/adapters/CardListAdapter.java]: you can use butterknife instead findviewbyid

File: MeetUp/app/src/main/res/layout/activity_group_info.xml (hunk 7 above)
Comment (pashacbk) [reply MeetUp/app/src/main/res/layout/activity_group_info.xml]: How I understood, this attribute make a stronger connection between layout and activity for better render and show styles in preview

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    private static final String GROUP_MEMBERS_STR1 = "We are ";
+    private static final String GROUP_MEMBERS_STR2 = " members here!";
+    private final GroupInfoActivityModel model;
+    private String description;
+
+    public GroupInfoActivityPresenter(GroupInfoActivityModel model) {
+        this.model = model;
+    }
+
+
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java]: extra space

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    private String transformDescription(Group group) {
+        description = group.getDescription();
+        description = description.replaceAll(HTML_BR_TAG, NEW_LINE_REPLACEMENT);
+        description = description.replaceAll(HTML_PARAGRAPH_START_TAG, TAB_STRING_REPLACEMENT);
+        return description = description.replaceAll(HTML_PARAGRAPH_END_TAG, NEW_LINE_REPLACEMENT);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java]: why you need this ?

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivity.java [hunk 11]
```diff
@@ -162,6 +163,17 @@ public void hideLoadingSwipeRefresh() {
...
 
+    @Override
+    public Observable<Integer> onGroupClick() {
+        return groupAdapter.onGroupClick();
+    }
+
+    @Override
+    public void startGroupInfoActivity(int groupId) {
+        startActivity(GroupInfoActivity.newIntent(this, groupId));
+
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivity.java]: extra line

File: MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java [hunk 12]
```diff
@@ -22,7 +22,7 @@
 
     private static final int FIRST_ELEMENT = 0;
     private static final String MY_GROUPS = "My Groups";
-    public static final String EMPTY_DATA_MESSAGE = "Seems that there is no stored data";
+    static final String EMPTY_DATA_MESSAGE = "Seems that there is no stored data";
```
Comment (UncuSergiu) [inline MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java]: i think this should be in resource

Comment (pashacbk) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/home/HomeActivityModel.java]: this changes are in Alexander's PR

File: MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java (hunk 10 above)
Comment (pashacbk) [reply MeetUp/app/src/main/java/com/endava/meetup/screens/group_screen/GroupInfoActivityPresenter.java]: will remove it

**Pull request merged**